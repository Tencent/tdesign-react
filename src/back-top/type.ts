/* eslint-disable */

/**
 * 该文件为脚本自动生成文件，请勿随意修改。如需修改请联系 PMC
 * */

import { TNode } from '../common';

export interface TdBackTopProps {
  /**
   * 是否绝对定位固定到屏幕右下方
   * @default true
   */
  fixed?: boolean;
  /**
   * 图标
   * @default 'backtop'
   */
  icon?: TNode;
  /**
   * 定位滚动到指定对象
   * @default () => window
   */
  target?: () => HTMLElement;
  /**
   * 文案
   * @default ''
   */
  text?: string;
  /**
   * 预设的样式类型
   * @default round
   */
  theme?: 'default' | 'primary' | 'dark';

  /**
   *  组件的形状
   */
  shape?: 'square' | 'circle';

  /**
   * 组件的大小
   */
  size?: 'small' | 'medium';
}
