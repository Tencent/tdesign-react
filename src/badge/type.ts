/* eslint-disable */

/**
 * 该文件为脚本自动生成文件，请勿随意修改。如需修改请联系 PMC
 * */

import { TNode } from '../common';

export interface TdBadgeProps {
  /**
   * 徽标内容，同 content
   */
  children?: TNode;
  /**
   * 颜色
   * @default ''
   */
  color?: string;
  /**
   * 徽标内容
   */
  content?: TNode;
  /**
   * 徽标右上角内容。可以是数字，也可以是文字。如：'new'/3/99+
   * @default 0
   */
  count?: TNode;
  /**
   * 是否为红点
   * @default false
   */
  dot?: boolean;
  /**
   * 封顶的数字值
   * @default 99
   */
  maxCount?: number;
  /**
   * 设置状态点的位置偏移，示例：[-10, 20] 或 ['10em', '8rem']
   */
  offset?: Array<string | number>;
  /**
   * 形状
   * @default circle
   */
  shape?: 'circle' | 'round';
  /**
   * 当数值为 0 时，是否展示徽标
   * @default false
   */
  showZero?: boolean;
  /**
   * 尺寸
   * @default medium
   */
  size?: 'small' | 'medium';
}
