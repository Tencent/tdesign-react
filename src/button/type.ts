/* eslint-disable */

/**
 * 该文件为脚本自动生成文件，请勿随意修改。如需修改请联系 PMC
 * */

import { TNode, TElement, SizeEnum } from '../common';
import { MouseEvent } from 'react';

export interface TdButtonProps {
  /**
   * 是否为块级元素
   * @default false
   */
  block?: boolean;
  /**
   * 按钮内容，同 content
   */
  children?: TNode;
  /**
   * 按钮内容
   */
  content?: TNode;
  /**
   * 是否禁用按钮
   * @default false
   */
  disabled?: boolean;
  /**
   * 是否为幽灵按钮（镂空按钮）
   * @default false
   */
  ghost?: boolean;
  /**
   * 按钮内部图标，可完全自定义
   */
  icon?: TElement;
  /**
   * 是否显示为加载状态
   * @default false
   */
  loading?: boolean;
  /**
   * 按钮形状，有 4 种：长方形、正方形、圆角长方形、圆形
   * @default rectangle
   */
  shape?: 'rectangle' | 'square' | 'round' | 'circle';
  /**
   * 组件尺寸
   * @default medium
   */
  size?: SizeEnum;
  /**
   * 组件风格，依次为默认色、品牌色、危险色、警告色、成功色
   */
  theme?: 'default' | 'primary' | 'danger' | 'warning' | 'success';
  /**
   * 按钮类型
   * @default button
   */
  type?: 'submit' | 'reset' | 'button';
  /**
   * 按钮形式，基础、线框、虚线、文字
   * @default base
   */
  variant?: 'base' | 'outline' | 'dashed' | 'text';
  /**
   * 点击时触发
   */
  onClick?: (e: MouseEvent<HTMLButtonElement>) => void;
}
