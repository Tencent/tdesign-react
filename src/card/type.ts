/* eslint-disable */

/**
 * 该文件为脚本自动生成文件，请勿随意修改。如需修改请联系 PMC
 * updated at 2021-12-27 17:08:43
 * */

import { TNode } from '../common';

export interface TdCardProps {
  /**
   * 卡片标题的操作区
   */
  actions?: TNode;
  /**
   * 是否有边框
   * @default true
   */
  bordered?: boolean;
  /**
   * 卡片封面图。值类型为字符串，会自动使用 `img` 标签输出封面图；也可以完全最定义封面图
   */
  cover?: TNode;
  /**
   * 卡片底部内容，可完全自定义
   */
  footer?: TNode;
  /**
   * 加载状态，值为 true 会根据不同的布局显示不同的加载状态，值为 false 则表示非加载状态。也可以使用 Sketon 组件完全自定义加载态呈现内容
   * @default false
   */
  loading?: TNode;
  /**
   * 是否显示卡片阴影，默认不显示
   * @default false
   */
  shadow?: boolean;
  /**
   * 尺寸
   * @default medium
   */
  size?: 'medium' | 'small';
  /**
   * 卡片标题
   */
  title?: TNode;
}
