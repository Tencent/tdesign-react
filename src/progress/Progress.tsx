import React, { forwardRef } from 'react';
import classNames from 'classnames';
import {
  CloseIcon as TdCloseIcon,
  CheckIcon as TdCheckIcon,
  ErrorIcon as TdErrorIcon,
  CheckCircleFilledIcon as TdCheckCircleFilledIcon,
  CloseCircleFilledIcon as TdCloseCircleFilledIcon,
  ErrorCircleFilledIcon as TdErrorCircleFilledIcon,
} from 'tdesign-icons-react';
import useConfig from '../hooks/useConfig';
import useGlobalIcon from '../hooks/useGlobalIcon';
import getBackgroundColor from '../_util/linearGradient';
import { StyledProps } from '../common';
import { TdProgressProps } from './type';
import { progressDefaultProps } from './defaultProps';

export interface ProgressProps extends TdProgressProps, StyledProps {}
/**
 * 按钮组件
 */
const Progress = forwardRef((props: ProgressProps, ref: React.Ref<HTMLDivElement>) => {
  const { classPrefix } = useConfig();
  const {
    CheckCircleIcon,
    CloseCircleIcon,
    ErrorCircleIcon,
    CheckCircleFilledIcon,
    CloseCircleFilledIcon,
    ErrorCircleFilledIcon,
  } = useGlobalIcon({
    CheckCircleIcon: TdCheckIcon,
    CloseCircleIcon: TdCloseIcon,
    ErrorCircleIcon: TdErrorIcon,
    CheckCircleFilledIcon: TdCheckCircleFilledIcon,
    CloseCircleFilledIcon: TdCloseCircleFilledIcon,
    ErrorCircleFilledIcon: TdErrorCircleFilledIcon,
  });

  const { theme, percentage, label, color = '', trackColor, strokeWidth, size, className } = props;
  let { status } = props;
  if (!status && percentage >= 100) {
    status = 'success';
  }
  let iconMap = {
    success: CheckCircleFilledIcon,
    error: CloseCircleFilledIcon,
    warning: ErrorCircleFilledIcon,
  };
  // 进度条展示内容
  const getInfoContent = () => {
    if (!label) {
      return '';
    }
    let info: React.ReactNode;
    // 为布尔值，默认百分百展示，否则之间展示 label 内容
    if (typeof label === 'boolean') {
      info = <div className={`${classPrefix}-progress__info`}>{`${percentage}%`}</div>;
      if (['success', 'error', 'warning'].includes(status)) {
        const Icon = iconMap[status];
        info = (
          <div className={`${classPrefix}-progress__info`}>
            <Icon className={`${classPrefix}-progress__icon`} />
          </div>
        );
      }
    } else {
      info = <div className={`${classPrefix}-progress__info`}>{label}</div>;
    }
    return info;
  };
  const statusClassName = `${classPrefix}-progress--status--${status}`;
  let progressDom;
  if (theme === 'circle') {
    iconMap = {
      success: CheckCircleIcon,
      error: CloseCircleIcon,
      warning: ErrorCircleIcon,
    };
    // 获取直径
    const getDiameter = (): number => {
      let diameter = 112;
      if (!size) {
        return diameter;
      }
      switch (size) {
        default:
          diameter = Number(size);
          break;
        case 'small':
          diameter = 72;
          break;
        case 'medium':
          diameter = 112;
          break;
        case 'large':
          diameter = 160;
          break;
      }
      return diameter;
    };
    // 获取环形进度条 环的宽度
    const getCircleStokeWidth = (): number => {
      if (!strokeWidth) {
        if (size === 'small') {
          return 4;
        }
      }
      if (typeof strokeWidth !== 'number' || Number.isNaN(strokeWidth)) {
        return 6;
      }
      return strokeWidth;
    };
    // 环形进度条尺寸(进度条占位空间，长宽占位)
    const circleStokeWidth = getCircleStokeWidth();
    // 直径
    const diameter = getDiameter();
    // 半径
    const radius = diameter / 2;
    // 内环半径
    const innerRadius = radius - circleStokeWidth;

    const perimeter = Math.PI * 2 * radius;
    const percent = percentage / 100;
    const strokeDasharray = `${perimeter * percent}  ${perimeter * (1 - percent)}`;
    // 自适应文字，根据半路，适度调整
    const fontSizeRatio = innerRadius * 0.27;
    const circleBoxStyle = {
      width: diameter,
      height: diameter,
      fontSize: 4 + fontSizeRatio,
    } as React.CSSProperties;
    const circlePathStyle = {
      stroke: color,
      strokeLinecap: circleStokeWidth < 30 ? 'round' : 'buff',
    } as React.CSSProperties;
    const circleCenterInViewBox = radius + circleStokeWidth / 2;
    progressDom = (
      <div
        ref={ref}
        className={classNames(`${classPrefix}-progress--circle`, {
          [`${statusClassName}`]: status,
        })}
        style={circleBoxStyle}
      >
        {getInfoContent()}
        <svg
          width={diameter}
          height={diameter}
          viewBox={`0 0 ${diameter + circleStokeWidth} ${diameter + circleStokeWidth}`}
        >
          <circle
            cx={circleCenterInViewBox}
            cy={circleCenterInViewBox}
            r={radius}
            strokeWidth={circleStokeWidth}
            stroke={trackColor || 'var(--td-bg-color-component)'}
            fill="none"
          ></circle>
          {percentage > 0 && (
            <circle
              cx={circleCenterInViewBox}
              cy={circleCenterInViewBox}
              r={radius}
              strokeWidth={circleStokeWidth}
              fill="none"
              transform={`matrix(0,-1,1,0,0,${diameter + circleStokeWidth})`}
              strokeDasharray={strokeDasharray}
              className={`${classPrefix}-progress__circle-inner`}
              style={circlePathStyle}
            ></circle>
          )}
        </svg>
      </div>
    );
    return <div className={className}>{progressDom}</div>;
  }
  // 进度条轨道高度
  const getHeight = (): string => {
    if (strokeWidth) {
      return typeof strokeWidth === 'string' ? strokeWidth : `${strokeWidth}px`;
    }
  };
  const trackStyle = {
    height: getHeight(),
    backgroundColor: trackColor,
    borderRadius: getHeight(),
  } as React.CSSProperties;
  const barStyle = {
    width: `${percentage}%`,
    background: getBackgroundColor(color),
    borderRadius: getHeight(),
  } as React.CSSProperties;
  if (theme === 'plump') {
    const PLUMP_SEPARATE = 10;
    progressDom = (
      <div
        ref={ref}
        className={classNames(`${classPrefix}-progress__bar`, `${classPrefix}-progress--plump`, {
          [`${statusClassName}`]: status,
          [`${classPrefix}-progress--over-ten`]: percentage > PLUMP_SEPARATE,
          [`${classPrefix}-progress--under-ten`]: percentage <= PLUMP_SEPARATE,
        })}
        style={trackStyle}
      >
        {percentage > PLUMP_SEPARATE ? (
          <div className={`${classPrefix}-progress__inner`} style={barStyle}>
            {label && (
              <div className={`${classPrefix}-progress__info`} style={{ color: '#fff' }}>{`${percentage}%`}</div>
            )}
          </div>
        ) : (
          <>
            <div className={`${classPrefix}-progress__inner`} style={barStyle}></div>
            {getInfoContent()}
          </>
        )}
      </div>
    );
  } else {
    progressDom = (
      <div
        ref={ref}
        className={classNames(`${classPrefix}-progress--thin`, {
          [`${statusClassName}`]: status,
        })}
      >
        <div className={`${classPrefix}-progress__bar`} style={trackStyle}>
          <div className={`${classPrefix}-progress__inner`} style={barStyle}></div>
        </div>
        {getInfoContent()}
      </div>
    );
  }
  return (
    <div className={className} style={props.style}>
      {progressDom}
    </div>
  );
});

Progress.displayName = 'Progress';
Progress.defaultProps = progressDefaultProps;

export default Progress;
