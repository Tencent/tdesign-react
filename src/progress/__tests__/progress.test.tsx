/* eslint-disable prefer-destructuring */
/*
 * @Author: Bin
 * @Date: 2022-04-07
 * @FilePath: /tdesign-react/src/progress/__tests__/progress.test.tsx
 */
import React from 'react';
import { render, testExamples, waitFor } from '@test/utils';
import Progress from '../Progress';
import { ThemeEnum } from '../type';

testExamples(__dirname);
describe('Progress 组件测试', () => {
  test('render theme', async () => {
    const testId = 'progress test theme';
    const themes: ThemeEnum[] = ['line', 'plump', 'circle'];
    const { getByTestId } = render(
      <div data-testid={testId}>
        {themes?.map((theme, index) => (
          <Progress key={index} strokeWidth={120} theme={theme} size={120} />
        ))}
      </div>,
    );

    const instance = await waitFor(() => getByTestId(testId));

    for (let index = 0; index < themes.length; index++) {
      const theme = themes[index];
      expect(() => instance.querySelector(`.t-progress--${theme}`)).not.toBe(null);
    }
  });

  test('render size', async () => {
    // const testId = 'progress test size';
    const sizes: any[] = [
      { name: 'small', size: 72 },
      { name: 'medium', size: 112 },
      { name: 'large', size: 160 },
      { name: 240, size: 240 },
    ];

    const createProgressSizeTest = async (sizeOjb: any) => {
      //   const sizeOjb = sizes[0];
      const testId = `progress test size ${sizeOjb.name}`;
      const view = render(
        <div data-testid={testId}>
          <Progress strokeWidth={'120px'} theme="circle" size={sizeOjb.name} />
        </div>,
      );
      const instance = await waitFor(() => view.getByTestId(testId));
      expect(instance.querySelector('.t-progress--circle')).toHaveStyle(`width: ${sizeOjb.size}px`);
    };

    await createProgressSizeTest(sizes[0]);
    await createProgressSizeTest(sizes[1]);
    await createProgressSizeTest(sizes[2]);
    await createProgressSizeTest(sizes[3]);
  });
});
