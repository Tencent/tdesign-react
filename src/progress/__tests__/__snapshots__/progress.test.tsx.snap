// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`circle.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-block-column-large"
    style="text-align: center;"
  >
    <div
      class="tdesign-demo-block-row"
    >
      <div
        class="tdesign-demo-block-column"
      >
        <div>
          默认样式
        </div>
        <div>
          <div
            class="t-progress--circle"
            style="width: 112px; height: 112px; font-size: 17.5px;"
          >
            <div
              class="t-progress__info"
            >
              0%
            </div>
            <svg
              height="112"
              viewBox="0 0 118 118"
              width="112"
            >
              <circle
                cx="59"
                cy="59"
                fill="none"
                r="56"
                stroke="var(--td-bg-color-component)"
                stroke-width="6"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="tdesign-demo-block-column"
      >
        <div>
          不显示数字
        </div>
        <div>
          <div
            class="t-progress--circle"
            style="width: 112px; height: 112px; font-size: 17.5px;"
          >
            <svg
              height="112"
              viewBox="0 0 118 118"
              width="112"
            >
              <circle
                cx="59"
                cy="59"
                fill="none"
                r="56"
                stroke="var(--td-bg-color-component)"
                stroke-width="6"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="tdesign-demo-block-column"
      >
        <div>
          自定义内容
        </div>
        <div>
          <div
            class="t-progress--circle"
            style="width: 112px; height: 112px; font-size: 17.5px;"
          >
            <div
              class="t-progress__info"
            >
              <div>
                75 day
              </div>
            </div>
            <svg
              height="112"
              viewBox="0 0 118 118"
              width="112"
            >
              <circle
                cx="59"
                cy="59"
                fill="none"
                r="56"
                stroke="var(--td-bg-color-component)"
                stroke-width="6"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tdesign-demo-block-row"
    >
      <div
        class="tdesign-demo-block-column"
      >
        <div>
          进度完成
        </div>
        <div>
          <div
            class="t-progress--circle t-progress--status--success"
            style="width: 112px; height: 112px; font-size: 17.5px;"
          >
            <div
              class="t-progress__info"
            >
              <svg
                class="t-icon t-icon-check-circle t-progress__icon"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M4.5 8.2L7 10.7l4.5-4.5-.7-.7L7 9.3 5.2 7.5l-.7.7z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
                <path
                  d="M4.11 2.18a7 7 0 117.78 11.64A7 7 0 014.1 2.18zm.56 10.8a6 6 0 106.66-9.97A6 6 0 004.67 13z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </div>
            <svg
              height="112"
              viewBox="0 0 118 118"
              width="112"
            >
              <circle
                cx="59"
                cy="59"
                fill="none"
                r="56"
                stroke="var(--td-bg-color-component)"
                stroke-width="6"
              />
              <circle
                class="t-progress__circle-inner"
                cx="59"
                cy="59"
                fill="none"
                r="56"
                stroke-dasharray="351.85837720205683  0"
                stroke-width="6"
                style="stroke-linecap: round;"
                transform="matrix(0,-1,1,0,0,118)"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="tdesign-demo-block-column"
      >
        <div>
          进度发生错误
        </div>
        <div>
          <div
            class="t-progress--circle t-progress--status--error"
            style="width: 112px; height: 112px; font-size: 17.5px;"
          >
            <div
              class="t-progress__info"
            >
              <svg
                class="t-icon t-icon-close-circle t-progress__icon"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M4.98 10.31L7.3 8 5 5.69l.7-.7L8 7.28 10.31 5l.7.7L8.72 8l2.3 2.31-.7.7L8 8.72 5.69 11l-.7-.7z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
                <path
                  d="M8 1a7 7 0 110 14A7 7 0 018 1zm0 1a6 6 0 100 12A6 6 0 008 2z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </div>
            <svg
              height="112"
              viewBox="0 0 118 118"
              width="112"
            >
              <circle
                cx="59"
                cy="59"
                fill="none"
                r="56"
                stroke="var(--td-bg-color-component)"
                stroke-width="6"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="tdesign-demo-block-column"
      >
        <div>
          进度被中断
        </div>
        <div>
          <div
            class="t-progress--circle t-progress--status--warning"
            style="width: 112px; height: 112px; font-size: 17.5px;"
          >
            <div
              class="t-progress__info"
            >
              <svg
                class="t-icon t-icon-error-circle t-progress__icon"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M8.5 4v5.5h-1V4h1zM8.6 10.5H7.4v1.2h1.2v-1.2z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
                <path
                  d="M15 8A7 7 0 101 8a7 7 0 0014 0zm-1 0A6 6 0 112 8a6 6 0 0112 0z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </div>
            <svg
              height="112"
              viewBox="0 0 118 118"
              width="112"
            >
              <circle
                cx="59"
                cy="59"
                fill="none"
                r="56"
                stroke="var(--td-bg-color-component)"
                stroke-width="6"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="tdesign-demo-block-column"
      >
        <div>
          自定义颜色
        </div>
        <div>
          <div
            class="t-progress--circle t-progress--status--error"
            style="width: 112px; height: 112px; font-size: 17.5px;"
          >
            <div
              class="t-progress__info"
            >
              <svg
                class="t-icon t-icon-close-circle t-progress__icon"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M4.98 10.31L7.3 8 5 5.69l.7-.7L8 7.28 10.31 5l.7.7L8.72 8l2.3 2.31-.7.7L8 8.72 5.69 11l-.7-.7z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
                <path
                  d="M8 1a7 7 0 110 14A7 7 0 018 1zm0 1a6 6 0 100 12A6 6 0 008 2z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </div>
            <svg
              height="112"
              viewBox="0 0 118 118"
              width="112"
            >
              <circle
                cx="59"
                cy="59"
                fill="none"
                r="56"
                stroke="#0f0"
                stroke-width="6"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tdesign-demo-block-row"
    >
      <div
        class="tdesign-demo-block-column"
      >
        <div>
          小尺寸
        </div>
        <div>
          <div
            class="t-progress--circle"
            style="width: 72px; height: 72px; font-size: 12.64px;"
          >
            <div
              class="t-progress__info"
            >
              0%
            </div>
            <svg
              height="72"
              viewBox="0 0 76 76"
              width="72"
            >
              <circle
                cx="38"
                cy="38"
                fill="none"
                r="36"
                stroke="var(--td-bg-color-component)"
                stroke-width="4"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="tdesign-demo-block-column"
      >
        <div>
          默认尺寸
        </div>
        <div>
          <div
            class="t-progress--circle"
            style="width: 112px; height: 112px; font-size: 17.5px;"
          >
            <div
              class="t-progress__info"
            >
              0%
            </div>
            <svg
              height="112"
              viewBox="0 0 118 118"
              width="112"
            >
              <circle
                cx="59"
                cy="59"
                fill="none"
                r="56"
                stroke="var(--td-bg-color-component)"
                stroke-width="6"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="tdesign-demo-block-column"
      >
        <div>
          大尺寸
        </div>
        <div>
          <div
            class="t-progress--circle"
            style="width: 160px; height: 160px; font-size: 23.98px;"
          >
            <div
              class="t-progress__info"
            >
              0%
            </div>
            <svg
              height="160"
              viewBox="0 0 166 166"
              width="160"
            >
              <circle
                cx="83"
                cy="83"
                fill="none"
                r="80"
                stroke="var(--td-bg-color-component)"
                stroke-width="6"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="tdesign-demo-block-column"
      >
        <div>
          自定义尺寸
        </div>
        <div>
          <div
            class="t-progress--circle"
            style="width: 112px; height: 112px; font-size: 5.62px;"
          >
            <div
              class="t-progress__info"
            >
              0%
            </div>
            <svg
              height="112"
              viewBox="0 0 162 162"
              width="112"
            >
              <circle
                cx="81"
                cy="81"
                fill="none"
                r="56"
                stroke="var(--td-bg-color-component)"
                stroke-width="50"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`line.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-demo-block-column"
  >
    <h3>
      默认在线形外展示进度和状态
    </h3>
    <div>
      默认样式
    </div>
    <div>
      <div
        class="t-progress--thin"
      >
        <div
          class="t-progress__bar"
        >
          <div
            class="t-progress__inner"
            style="width: 0%;"
          />
        </div>
        <div
          class="t-progress__info"
        >
          0%
        </div>
      </div>
    </div>
    <div>
      进度被中断
    </div>
    <div>
      <div
        class="t-progress--thin t-progress--status--warning"
      >
        <div
          class="t-progress__bar"
        >
          <div
            class="t-progress__inner"
            style="width: 0%;"
          />
        </div>
        <div
          class="t-progress__info"
        >
          <svg
            class="t-icon t-icon-error-circle-filled t-progress__icon"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M15 8A7 7 0 101 8a7 7 0 0014 0zM8.5 4v5.5h-1V4h1zm-1.1 7h1.2v1.2H7.4V11z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
      </div>
    </div>
    <div>
      进度状态发生重大错误
    </div>
    <div>
      <div
        class="t-progress--thin t-progress--status--error"
      >
        <div
          class="t-progress__bar"
        >
          <div
            class="t-progress__inner"
            style="width: 0%;"
          />
        </div>
        <div
          class="t-progress__info"
        >
          <svg
            class="t-icon t-icon-close-circle-filled t-progress__icon"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M15 8A7 7 0 101 8a7 7 0 0014 0zM5.67 4.95L8 7.29l2.33-2.34.7.7L8.7 8l2.34 2.35-.71.7L8 8.71l-2.33 2.34-.7-.7L7.3 8 4.96 5.65l.71-.7z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
      </div>
    </div>
    <div>
      进度正常更新
    </div>
    <div>
      <div
        class="t-progress--thin t-progress--status--active"
      >
        <div
          class="t-progress__bar"
        >
          <div
            class="t-progress__inner"
            style="width: 0%;"
          />
        </div>
        <div
          class="t-progress__info"
        >
          0%
        </div>
      </div>
    </div>
    <div>
      不显示数字
    </div>
    <div>
      <div
        class="t-progress--thin"
      >
        <div
          class="t-progress__bar"
        >
          <div
            class="t-progress__inner"
            style="width: 0%;"
          />
        </div>
      </div>
    </div>
    <div>
      自定义内容
    </div>
    <div>
      <div
        class="t-progress--thin"
      >
        <div
          class="t-progress__bar"
        >
          <div
            class="t-progress__inner"
            style="width: 0%;"
          />
        </div>
        <div
          class="t-progress__info"
        >
          <div>
            自定义文本
          </div>
        </div>
      </div>
    </div>
    <div>
      自定义颜色与高度
    </div>
    <div>
      <div
        class="t-progress--thin"
      >
        <div
          class="t-progress__bar"
          style="height: 30px; background-color: rgb(0, 255, 0); border-radius: 30px;"
        >
          <div
            class="t-progress__inner"
            style="width: 0%; background: rgb(0, 0, 255); border-radius: 30px;"
          />
        </div>
        <div
          class="t-progress__info"
        >
          0%
        </div>
      </div>
    </div>
    <div>
      进度条渐变色
    </div>
    <div>
      <div
        class="t-progress--thin"
      >
        <div
          class="t-progress__bar"
        >
          <div
            class="t-progress__inner"
            style="width: 0%;"
          />
        </div>
        <div
          class="t-progress__info"
        >
          0%
        </div>
      </div>
    </div>
    <div>
      <div
        class="t-progress--thin"
      >
        <div
          class="t-progress__bar"
          style="background-color: rgb(0, 255, 0);"
        >
          <div
            class="t-progress__inner"
            style="width: 0%;"
          />
        </div>
        <div
          class="t-progress__info"
        >
          0%
        </div>
      </div>
    </div>
    <div>
      <div
        class="t-progress--thin"
      >
        <div
          class="t-progress__bar"
          style="height: 30px; border-radius: 30px;"
        >
          <div
            class="t-progress__inner"
            style="width: 0%; border-radius: 30px;"
          />
        </div>
        <div
          class="t-progress__info"
        >
          0%
        </div>
      </div>
    </div>
    <h3>
      可以在线形内展示进度信息
    </h3>
    <div
      class="tdesign-demo-block-column"
    >
      <div>
        默认样式
      </div>
      <div>
        <div
          class="t-progress__bar t-progress--plump t-progress--over-ten"
        >
          <div
            class="t-progress__inner"
            style="width: 30%;"
          >
            <div
              class="t-progress__info"
              style="color: rgb(255, 255, 255);"
            >
              30%
            </div>
          </div>
        </div>
      </div>
      <div>
        进度0-10%时数字数字位置出现在目前进度的右边区域
      </div>
      <div>
        <div
          class="t-progress__bar t-progress--plump t-progress--under-ten"
        >
          <div
            class="t-progress__inner"
            style="width: 5%;"
          />
          <div
            class="t-progress__info"
          >
            5%
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`plump.jsx 1`] = `
<DocumentFragment>
  <div>
    <div
      style="margin: 20px 0px 10px;"
    >
      默认样式
    </div>
    <div>
      <div
        class="t-progress__bar t-progress--plump t-progress--over-ten"
      >
        <div
          class="t-progress__inner"
          style="width: 30%;"
        >
          <div
            class="t-progress__info"
            style="color: rgb(255, 255, 255);"
          >
            30%
          </div>
        </div>
      </div>
    </div>
    <div
      style="margin: 20px 0px 10px;"
    >
      进度0-10%时数字数字位置出现在目前进度的右边区域
    </div>
    <div>
      <div
        class="t-progress__bar t-progress--plump t-progress--under-ten"
      >
        <div
          class="t-progress__inner"
          style="width: 5%;"
        />
        <div
          class="t-progress__info"
        >
          5%
        </div>
      </div>
    </div>
    <div
      style="margin: 20px 0px 10px;"
    >
      100%
    </div>
    <div>
      <div
        class="t-progress__bar t-progress--plump t-progress--status--success t-progress--over-ten"
      >
        <div
          class="t-progress__inner"
          style="width: 100%;"
        >
          <div
            class="t-progress__info"
            style="color: rgb(255, 255, 255);"
          >
            100%
          </div>
        </div>
      </div>
    </div>
    <div
      style="margin: 20px 0px 10px;"
    >
      success
    </div>
    <div>
      <div
        class="t-progress__bar t-progress--plump t-progress--status--success t-progress--over-ten"
      >
        <div
          class="t-progress__inner"
          style="width: 100%;"
        >
          <div
            class="t-progress__info"
            style="color: rgb(255, 255, 255);"
          >
            100%
          </div>
        </div>
      </div>
    </div>
    <div
      style="margin: 20px 0px 10px;"
    >
      warning
    </div>
    <div>
      <div
        class="t-progress__bar t-progress--plump t-progress--status--warning t-progress--over-ten"
      >
        <div
          class="t-progress__inner"
          style="width: 30%;"
        >
          <div
            class="t-progress__info"
            style="color: rgb(255, 255, 255);"
          >
            30%
          </div>
        </div>
      </div>
    </div>
    <div
      style="margin: 20px 0px 10px;"
    >
      error
    </div>
    <div>
      <div
        class="t-progress__bar t-progress--plump t-progress--status--error t-progress--over-ten"
      >
        <div
          class="t-progress__inner"
          style="width: 30%;"
        >
          <div
            class="t-progress__info"
            style="color: rgb(255, 255, 255);"
          >
            30%
          </div>
        </div>
      </div>
    </div>
    <div
      style="margin: 20px 0px 10px;"
    >
      active
    </div>
    <div>
      <div
        class="t-progress__bar t-progress--plump t-progress--status--active t-progress--over-ten"
      >
        <div
          class="t-progress__inner"
          style="width: 30%;"
        >
          <div
            class="t-progress__info"
            style="color: rgb(255, 255, 255);"
          >
            30%
          </div>
        </div>
      </div>
    </div>
    <div
      style="margin: 20px 0px 10px;"
    >
      不显示数字
    </div>
    <div>
      <div
        class="t-progress__bar t-progress--plump t-progress--over-ten"
      >
        <div
          class="t-progress__inner"
          style="width: 30%;"
        />
      </div>
    </div>
    <div
      style="margin: 20px 0px 10px;"
    >
      自定义颜色与尺寸
    </div>
    <div>
      <div
        class="t-progress__bar t-progress--plump t-progress--over-ten"
        style="height: 30px; background-color: rgb(0, 255, 0); border-radius: 30px;"
      >
        <div
          class="t-progress__inner"
          style="width: 30%; background: rgb(0, 0, 255); border-radius: 30px;"
        >
          <div
            class="t-progress__info"
            style="color: rgb(255, 255, 255);"
          >
            30%
          </div>
        </div>
      </div>
    </div>
    <div
      style="margin: 20px 0px 10px;"
    >
      进度条渐变色
    </div>
    <div>
      <div
        class="t-progress__bar t-progress--plump t-progress--over-ten"
      >
        <div
          class="t-progress__inner"
          style="width: 30%;"
        >
          <div
            class="t-progress__info"
            style="color: rgb(255, 255, 255);"
          >
            30%
          </div>
        </div>
      </div>
    </div>
    <br />
    <div>
      <div
        class="t-progress__bar t-progress--plump t-progress--over-ten"
        style="background-color: rgb(0, 255, 0);"
      >
        <div
          class="t-progress__inner"
          style="width: 30%;"
        >
          <div
            class="t-progress__info"
            style="color: rgb(255, 255, 255);"
          >
            30%
          </div>
        </div>
      </div>
    </div>
    <br />
    <div>
      <div
        class="t-progress__bar t-progress--plump t-progress--over-ten"
      >
        <div
          class="t-progress__inner"
          style="width: 30%;"
        >
          <div
            class="t-progress__info"
            style="color: rgb(255, 255, 255);"
          >
            30%
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
